pkgs/by-name/do/dorion/package.nix

```diff
@@ -1,65 +1,192 @@
-{ lib
-, stdenv
-, fetchurl
-, autoPatchelfHook
-, dpkg
-, glib-networking
-, gst_all_1
-, libappindicator
-, libayatana-appindicator
-, webkitgtk_4_0
-, wrapGAppsHook3
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchurl,
+  rustPlatform,
+  autoPatchelfHook,
+  wrapGAppsHook3,
+  glib-networking,
+  gst_all_1,
+  libappindicator,
+  libayatana-appindicator,
+  nodejs,
+  openssl,
+  pkg-config,
+  yq-go,
+  pnpm,
+  webkitgtk_4_1,
+  makeDesktopItem,
+  copyDesktopItems,
 }:
 
-stdenv.mkDerivation (finalAttrs: {
+let
   pname = "dorion";
-  version = "5.0.1";
+  version = "6.2.0";
 
-  src = fetchurl {
-    url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version }/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-cCZikTZ+IU3mq/FkJfeggXLyWIsWG+a2qu1GbgW93WQ=";
+  webkitgtk_4_1' = webkitgtk_4_1.override { enableExperimental = true; };
+
+  src = fetchFromGitHub {
+    owner = "SpikeHD";
+    repo = "Dorion";
+    rev = "v${version}";
+    hash = "sha256-7isOL/853IMOUkepbu81INNFgjTHpo9dyvPaxMkBU1U=";
+
+    nativeBuildInputs = [ pnpm.configHook ];
   };
 
-  unpackCmd = ''
-    dpkg -X $curSrc .
-  '';
+  shelter = fetchurl {
+    url = "https://raw.githubusercontent.com/uwu/shelter-builds/005fd07996b8f2786409e5a7e1418763a96f55d9/shelter.js";
+    hash = "sha256-eFR/974pUP2o/A3FyOIeQbD2tVbMPpdGCoCuk4ZhZos=";
+    meta = {
+      homepage = "https://github.com/uwu/shelter";
+      sourceProvenance = [ lib.sourceTypes.binaryBytecode ]; # actually, minified JS
+      license = lib.licenses.cc0;
+    };
+  };
 
-  runtimeDependencies = [
-    glib-networking
-    libappindicator
-    libayatana-appindicator
-  ];
+  frontend = stdenv.mkDerivation {
+    pname = "dorion-ui";
+    inherit version src;
+
+    pnpmDeps = pnpm.fetchDeps {
+      inherit pname version src;
+      hash = "sha256-Exl70+eeGpzn84Wt4RzEfA12fVASuJkVXYgR1dvXL3U=";
```

Comment:
Please rebase and regenerate hashes.

See https://github.com/NixOS/nixpkgs/pull/349093#issuecomment-2421873637

Reply:
Done package still broken though.